cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -91,51 +201,730 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
     throws SchedulerException {
-    return ImmutableList.of();
+    char startChar = ':';
+    char endChar = (char) (startChar + 1);
+    String programScheduleId = AbstractSchedulerService.programIdFor(program, programType);
+    return ImmutableList.copyOf(scheduleSubscribers.subMap(String.format("%s%c", programScheduleId, startChar),
+                                                           String.format("%s%c", programScheduleId, endChar))
+                                  .keySet());
   }
 
   @Override
   public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
-    throws NotFoundException, SchedulerException {
-
+    throws ScheduleNotFoundException, SchedulerException {
+    String scheduleId = AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName);
+    StreamSubscriber subscriber = scheduleSubscribers.get(scheduleId);
+    if (subscriber == null) {
+      throw new ScheduleNotFoundException(scheduleName);
+    }
+    subscriber.suspendScheduleTask(program, programType, scheduleName);
   }
 
   @Override
   public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
-    throws NotFoundException, SchedulerException {
-
+    throws ScheduleNotFoundException, SchedulerException {
+    String scheduleId = AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName);
+    StreamSubscriber subscriber = scheduleSubscribers.get(scheduleId);
+    if (subscriber == null) {
+      throw new ScheduleNotFoundException(scheduleName);
+    }
+    subscriber.resumeScheduleTask(program, programType, scheduleName);
   }
 
   @Override
   public void updateSchedule(Id.Program program, SchedulableProgramType programType, Schedule schedule)
     throws NotFoundException, SchedulerException {
+    Preconditions.checkArgument(schedule instanceof StreamSizeSchedule,
+                                "Schedule should be of type StreamSizeSchedule");
+    StreamSizeSchedule streamSizeSchedule = (StreamSizeSchedule) schedule;
+    StreamSubscriber subscriber = scheduleSubscribers.get(AbstractSchedulerService.scheduleIdFor(program, programType,
+                                                                                                 schedule.getName()));
+    if (subscriber == null) {
+      throw new ScheduleNotFoundException(schedule.getName());
+    }
 
+    if (!streamSizeSchedule.getStreamName().equals(subscriber.getStreamId().getName())) {
+      // For a change of stream, it's okay to delete the schedule and recreate it
+      try {
+        deleteSchedule(program, programType, schedule.getName());
+      } catch (NotFoundException e) {
+        // It can happen that the schedule was deleted while being updated. In which case, the update action
+        // came first and we still want to create it
+        LOG.warn("Schedule {} deleted while being updated", schedule.getName(), e);
```

Comment:
the comment is wrong. Delete deletes the schedule, regardless of a race with update.